src/rpc/format/cbor/convert.rs

```diff
@@ -263,9 +279,12 @@ impl TryFrom<Value> for Cbor {
 				Ok(Cbor(Data::Tag(TAG_DURATION, Box::new(Data::Text(v.to_raw())))))
 			}
 			Value::Datetime(v) => {
-				Ok(Cbor(Data::Tag(TAG_DATETIME, Box::new(Data::Text(v.to_raw())))))
+				Ok(Cbor(Data::Tag(TAG_DATETIME, Box::new(Data::Array(vec![
+					Data::Integer(v.0.timestamp().into()),
+					Data::Integer(v.0.timestamp_subsec_nanos().into()),
+				])))))
```

Comment:
What will happen in languages which don't support nanosecond precision?

Reply:
As it is already a custom cbor tag, it does not matter. 

Note that C# does not have a notion of nanosecond, only `Tick`. So, as expected, it is up to the client to handle the transformation.

Reply:
Noting here that this will technically be a breaking change, unfortunately

Reply:
I actually like this solution better.